pool.go

```diff
@@ -0,0 +1,99 @@
+// Copyright 2016 Canonical Ltd.
+// Licensed under the LGPLv3, see LICENCE file for details.
+
+package gomaasapi
+
+import (
+	"github.com/juju/errors"
+	"github.com/juju/schema"
+	"github.com/juju/version"
+)
+
+type pool struct {
+	// Add the controller in when we need to do things with the pool.
+	// controller Controller
+
+	resourceURI string
+
+	name        string
+	description string
+}
+
+// Name implements Pool.
+func (p *pool) Name() string {
+	return p.name
+}
+
+// Description implements Pool.
+func (p *pool) Description() string {
+	return p.description
+}
+
+func readPools(controllerVersion version.Number, source interface{}) ([]*pool, error) {
+	checker := schema.List(schema.StringMap(schema.Any()))
+	coerced, err := checker.Coerce(source, nil)
+	if err != nil {
+		return nil, errors.Annotatef(err, "pool base schema check failed")
+	}
+	valid := coerced.([]interface{})
+
+	var deserialisationVersion version.Number
+	for v := range poolDeserializationFuncs {
+		if v.Compare(deserialisationVersion) > 0 && v.Compare(controllerVersion) <= 0 {
+			deserialisationVersion = v
+		}
+	}
+	if deserialisationVersion == version.Zero {
+		return nil, errors.Errorf("no pool read func for version %s", controllerVersion)
+	}
+	readFunc := poolDeserializationFuncs[deserialisationVersion]
+	return readPoolList(valid, readFunc)
+}
+
+// readPoolList expects the values of the sourceList to be string maps.
+func readPoolList(sourceList []interface{}, readFunc poolDeserializationFunc) ([]*pool, error) {
+	result := make([]*pool, 0, len(sourceList))
+	for i, value := range sourceList {
+		source, ok := value.(map[string]interface{})
+		if !ok {
+			return nil, errors.Errorf("unexpected value for pool %d, %T", i, value)
+		}
+		pool, err := readFunc(source)
+		if err != nil {
+			return nil, errors.Annotatef(err, "pool %d", i)
+		}
+		result = append(result, pool)
+	}
+	return result, nil
+}
+
+type poolDeserializationFunc func(map[string]interface{}) (*pool, error)
+
+var poolDeserializationFuncs = map[version.Number]poolDeserializationFunc{
+	twoDotOh: Pool2dot0,
+}
+
+func Pool2dot0(source map[string]interface{}) (*pool, error) {
```

Comment:
this doesn't need to be exported, so rename to pool_2_0
(to be consistent with zone)